### PR TITLE
Add audience column to groups-role and user-role page

### PR DESCRIPTION
**Pull request by kayathiri4**
### Related Issues
**Groups Page** - https://github.com/wso2/product-is/issues/17650
Screenshot from 2023-11-17 17-50-22

**Users Page** - https://github.com/wso2/product-is/issues/17648
Screenshot from 2023-11-17 18-12-43

### Related PRs
- None

### Checklist
- [ ] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

**Comment by pavinduLakshan** [comment]
Add changeset

File: modules/react-components/src/components/transfer/transfer-list-item.tsx [hunk 1]
```diff
@@ -116,24 +163,12 @@ export const TransferListItem: FunctionComponent<TransferListItemPropsInterface>
...
-                        />
-                    </Table.Cell>
-                )
+                !reOrderLabel && renderLabel()
             }
             {
                 showListSubItem
                     ? (
-                        <Table.Cell id={ listItemId }>
+                        <Table.Cell id={ listItemId } >
```
Comment (pavinduLakshan) [inline modules/react-components/src/components/transfer/transfer-list-item.tsx:171]:
```suggestion
                        <Table.Cell id={ listItemId }>
```

File: apps/console/src/features/core/components/roles/assign-roles.tsx [hunk 2]
```diff
@@ -257,10 +257,14 @@ export const AssignRoles: FunctionComponent<AssignRoleProps> = (props: AssignRol
...
                                         listItemId={ role.id }
                                         listItemIndex={ index }
-                                        listItemTypeLabel={ createItemLabel(role?.displayName) }
+                                        listItemTypeLabel={ 
+                                            createItemLabel(role?.audience.type, role?.audience.display) 
+                                        }
                                         isItemChecked={ checkedUnassignedListItems.includes(role) }
                                         showSecondaryActions={ false }
                                         handleOpenPermissionModal={ () => handleSetRoleId(role.id) }
+                                        reOrderLabel={ true }
```
Comment (pavinduLakshan) [inline apps/console/src/features/core/components/roles/assign-roles.tsx:266]:
```suggestion
                                        reOrderLabel
                                        showSubLabel
```

File: modules/react-components/src/components/transfer/transfer-list-item.tsx [hunk 3]
```diff
@@ -102,8 +107,50 @@ export const TransferListItem: FunctionComponent<TransferListItemPropsInterface>
...
+                        )
+                    )
+                }
+            </>
+        );
+    };
+
     return (
-        <Table.Row key={ listItemIndex }>
+        <Table.Row key={ listItemIndex } >
```
Comment (pavinduLakshan) [inline modules/react-components/src/components/transfer/transfer-list-item.tsx:153]:
```suggestion
        <Table.Row key={ listItemIndex }>
```

File: apps/console/src/features/core/components/roles/assign-roles.tsx [hunk 4]
```diff
@@ -200,16 +200,16 @@ export const AssignRoles: FunctionComponent<AssignRoleProps> = (props: AssignRol
...
-        const role: string[] = roleName.split("/");
-
+    const createItemLabel = (roleType: string, application: string) => {
+        const role: string[] = roleType.split("/");
+        
         if (role.length > 0) {
-            if (role[0] == "Application") {
-                return { labelColor: null, labelText: "Application", name: "application-label" };
+            if (role[0] == "application") {
+                return { labelColor: null, labelText: "Application", name: "audience-label", subLabel: application };
```
Comment (brionmario) [inline apps/console/src/features/core/components/roles/assign-roles.tsx:210]: Can't we use i18n for `labelText`?

**Review by brionmario — DISMISSED** [review]

**Review by brionmario — APPROVED** [review]

**Pull request merged**